base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] Setters and getters for the instance variable `@namebale` should be present.